main.tf

```
@@ -603,3 +603,26 @@ resource "aws_iam_instance_profile" "this" {
     create_before_destroy = true
   }
 }
+
+################################################################################
+# Elastic IP
+################################################################################
+
+resource "aws_eip" "public" {
```

```suggestion
resource "aws_eip" "this" {
```

Why call it "public"? Can it be of another type?